MetaMask/metamask-extension
# Ability to buy tokens with Coinbase Pay and Transak

## Explanation

Current State: Only native currencies can be bought using on ramp through the buy button
After Change: Ability to purchase tokens defined in `BUYABLE_CHAINS_MAP` using Coinbase Pay and Transak

Changes:

- migrated old buy eth modal to a new one using the new popover component
- new modal now supports buying both native currencies and tokens
- if a token is defined as buyable on the `BUYABLE_CHAINS_MAP` on at least one provider, a buy button will show on the token overview page



## More Information

See ONRAMP-189 and ONRAMP-190



## Screenshots/Screencaps



### Before
![Screenshot 2022-09-07 at 16 33 19](https://user-images.githubusercontent.com/8658960/188905816-416b8d06-b0ec-4a5e-a91a-d26e2a7e5522.png)




### After
![Screenshot 2022-09-07 at 16 37 25](https://user-images.githubusercontent.com/8658960/188905966-212ca576-26e6-49c4-9929-4132edfb2a64.png)
![Screenshot 2022-09-07 at 16 33 43](https://user-images.githubusercontent.com/8658960/188906000-94a4175d-e4b6-41bb-8b8b-452f91124f66.png)
![Screenshot 2022-09-07 at 16 34 11](https://user-images.githubusercontent.com/8658960/188906016-c25baf0a-040e-4d6f-92ad-44ee26be2963.png)




## Manual Testing Steps

These changes apply only to Ethereum mainnet

- Import a token
- If token is on the [Transak list](https://github.com/MetaMask/metamask-extension/pull/15551/commits/ffbface1c125c1b9df03f1c33f80133879415673#diff-4e682f9ffe2eed5743a5d0b761ffdab503afaad448954bd69ce2f7c8bca000d1R266) or on the [Coinbase Pay list](https://github.com/MetaMask/metamask-extension/pull/15551/commits/ffbface1c125c1b9df03f1c33f80133879415673#diff-4e682f9ffe2eed5743a5d0b761ffdab503afaad448954bd69ce2f7c8bca000d1R309), it should display a "Buy" button
- Depending on which list it shows, it should give you the option to buy the token on said provider
- Once you click on the provider's "Continue to" button, a window on said provider should open, and the token to buy should already be preselected



## Pre-Merge Checklist


### + If there are functional changes:
